### DISCUSSION TITLE
Viewing Snowflake data based on logged in user account

### DISCUSSION THREAD

**Discussion by arunconfig** [apache/superset#28519] [General]
In Power BI, I can build dashboards using direct query to Snowflake and publish a report to Power BI service. And this will render the data based on the logged in users access in Snowflake, as we have turned on Snowflake SSO in power bi service. 

If we configure Apache superset to use SSO for its authentication, then establish a connection to Snowflake database using a service account -> build dashboard -> publish it to end users. Will or Can the Snowflake data be restricted based on end user's AD account? I am not able to find anything equivalent to Snowflake SSO Power bi functionality in apache superset.  

https://www.snowflake.com/blog/using-sso-between-power-bi-and-snowflake/

**Comment by ablanchard** [comment]
We are confronted with the same problem using databricks. Our superset has a connection to databricks using a Service Account, and we haven't found a way to use an account based on the superset user connected. 
The way, the database is configured in superset makes it impossible, for the moment I think. But would be a very nice feature to have, in term of data governance.

**Comment by ablanchard** [comment]
In fact, by searching in the superset slack community, there is this way to do it. If you have a user connected in superset, you can use `DB_CONNECTION_MUTATOR` to customize the token used for connecting to your snowflake. This thread gives insights on how to do it https://github.com/apache/superset/issues/17074
